rt/emul/mini/src/main/resources/org/apidesign/vm4brwsr/emul/lang/java_lang_Number.js

```diff
@@ -1,572 +1,2671 @@
+
+  var encodedName = "[" + componentData.arrayEncodedName;
+  var componentBase = componentData.arrayBase || componentData;
+  var arrayDepth = componentData.arrayDepth + 1;
+
+  var isInstance = function(obj) {
+    return componentBase.isArrayOf(obj, arrayDepth);
+  }
+
+  // Runtime support
+  this.constr = ArrayClass;
+  this.parentData = $d_O;
+  this.ancestors = {O: 1, jl_Cloneable: 1, Ljava_io_Serializable: 1};
+  this.componentData = componentData;
+  this.arrayBase = componentBase;
+  this.arrayDepth = arrayDepth;
+  this.zero = null;
+  this.arrayEncodedName = encodedName;
+  this._classOf = undefined;
+  this._arrayOf = undefined;
+  this.isArrayOf = undefined;
+
+  // java.lang.Class support
+  this["name"] = encodedName;
+  this["isPrimitive"] = false;
+  this["isInterface"] = false;
+  this["isArrayClass"] = true;
+  this["isInstance"] = isInstance;
+
+  return this;
+};
+
+
+$TypeData.prototype.getClassOf = function() {
+
+
+
+  if (!this._classOf)
+    this._classOf = new $c_jl_Class().init___jl_ScalaJSClassData(this);
+  return this._classOf;
+};
+
+
+$TypeData.prototype.getArrayOf = function() {
+
+
+
+  if (!this._arrayOf)
+    this._arrayOf = new $TypeData().initArray(this);
+  return this._arrayOf;
+};
+
+// java.lang.Class support
+
+
+$TypeData.prototype["getFakeInstance"] = function() {
+
+
+
+  if (this === $d_T)
+    return "some string";
+  else if (this === $d_jl_Boolean)
+    return false;
+  else if (this === $d_jl_Byte ||
+           this === $d_jl_Short ||
+           this === $d_jl_Integer ||
+           this === $d_jl_Float ||
+           this === $d_jl_Double)
+    return 0;
+  else if (this === $d_jl_Long)
+    return $m_sjsr_RuntimeLong$().Zero$1;
+  else if (this === $d_sr_BoxedUnit)
+    return void 0;
+  else
+    return {$classData: this};
+};
+
+
+$TypeData.prototype["getSuperclass"] = function() {
+
+
+
+  return this.parentData ? this.parentData.getClassOf() : null;
+};
+
+
+$TypeData.prototype["getComponentType"] = function() {
+
+
+
+  return this.componentData ? this.componentData.getClassOf() : null;
+};
+
+
+$TypeData.prototype["newArrayOfThisClass"] = function(lengths) {
+
+
+
+  var arrayClassData = this;
+  for (var i = 0; i < lengths.length; i++)
+    arrayClassData = arrayClassData.getArrayOf();
+  return $newArrayObject(arrayClassData, lengths);
+};
+
+
+
+
+// Create primitive types
+
+var $d_V = new $TypeData().initPrim(undefined, "V", "void");
+var $d_Z = new $TypeData().initPrim(false, "Z", "boolean");
+var $d_C = new $TypeData().initPrim(0, "C", "char");
+var $d_B = new $TypeData().initPrim(0, "B", "byte");
+var $d_S = new $TypeData().initPrim(0, "S", "short");
+var $d_I = new $TypeData().initPrim(0, "I", "int");
+var $d_J = new $TypeData().initPrim("longZero", "J", "long");
+var $d_F = new $TypeData().initPrim(0.0, "F", "float");
+var $d_D = new $TypeData().initPrim(0.0, "D", "double");
+
+// Instance tests for array of primitives
+
+var $isArrayOf_Z = $makeIsArrayOfPrimitive($d_Z);
```

Comment:
All `$isArrayXYZ` functions are just generated, but never used (except in the RTTI part).

Reply:
Yes, those are part of [the hand-written JS code](https://github.com/scala-js/scala-js/blob/v1.0.0-M3/linker/scalajsenv.js) which is not analyzed for actual usage. Normally we rely on Closure in `fullOptJS` to get rid of those little details. It is good at dealing with top-level functions (less so at dealing with methods in hierarchies, which is why we do that ourselves).